src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Tests/AdminServiceTest.cs

```diff
@@ -1,145 +1,154 @@
+            var test = GetTest();
             this._mapper = new MapperConfiguration(cfg => cfg.AddProfile<TestProfile>()).CreateMapper();
-            var adminService = new AdminService(this._adminRepositoryMock.Object, this._baseTestRepositoryMock.Object, this._mapper, null);
+            var adminService = new AdminService(this._testRepositoryMock.Object, this._testAnswerRepositoryMock.Object, this._mapper, null);
 
             // Act
-            adminService.UpdateTestDates(this._testDto, "20");
+            adminService.UpdateTestDates(test.Id, "20");
 
             // Assert
-            this._adminRepositoryMock.Verify(mock => mock.UpdateDates(It.Is<Test>(x => x.Number == this._testDto.Number && x.Id == this._testDto.Id && x.StartDate == this._testDto.StartDate && x.EndDate == this._testDto.EndDate)), Times.Once());
+            this._testRepositoryMock.Verify(mock => mock.UpdateDates(test.Id, It.IsAny<DateTime>(), It.IsAny<DateTime>()), Times.Once());
         }
 
         [Fact]
         public void UpdateTestEndDate()
         {
             // Arrange
-            this._adminRepositoryMock.Setup(mock => mock.UpdateEndDate(It.IsAny<Test>()));
+            var test = GetTest();
+            var newDate = DateTime.Today.AddHours(23).AddMinutes(20);
             this._mapper = new MapperConfiguration(cfg => cfg.AddProfile<TestProfile>()).CreateMapper();
-            var adminService = new AdminService(this._adminRepositoryMock.Object, this._baseTestRepositoryMock.Object, this._mapper, null);
+            var adminService = new AdminService(this._testRepositoryMock.Object, this._testAnswerRepositoryMock.Object, this._mapper, null);
 
             // Act
-            adminService.UpdateTestEndDate(this._testDto, (DateTime)this._testDto.EndDate);
+            adminService.UpdateTestEndDate(test.Id, newDate);
 
             // Assert
-            this._adminRepositoryMock.Verify(mock => mock.UpdateEndDate(It.Is<Test>(x => x.Number == this._testDto.Number && x.Id == this._testDto.Id && x.StartDate == this._testDto.StartDate && x.EndDate == this._testDto.EndDate)), Times.Once());
+            this._testRepositoryMock.Verify(mock => mock.UpdateEndDate(test.Id, newDate), Times.Once());
         }
+
+        private static List<Test> GetTestList() => new List<Test>()
+        {
+            new Test()
+            {
+                Id = 1,
+                Number = 1,
+                StartDate = DateTime.Today.AddHours(12),
+                EndDate = DateTime.Today.AddHours(23).AddMinutes(59),
+                Answers = null
+            },
+            new Test()
+            {
+                Id = 2,
+                Number = 2,
+                StartDate = DateTime.Today.AddHours(10),
+                EndDate = DateTime.Today.AddHours(22).AddMinutes(0),
+                Answers = null
+            }
+        };
+
+        private static Test GetTest() => new Test()
+        {
+            Id = 2,
```

Comment:
I suggest adding the parameter `number` and use it in properties `Id` and `Number`